src/Core/Domain/CartRule/ValueObject/PercentageDiscount.php

```diff
@@ -26,50 +26,48 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\CartRule\ValueObject;
 
-use PrestaShop\PrestaShop\Core\Domain\CartRule\Exception\CartRuleConstraintException;
+use PrestaShop\Decimal\DecimalNumber;
 
 /**
  * Percentage discount
  */
 class PercentageDiscount
 {
     /**
-     * @var float
+     * @var DecimalNumber
      */
     private $percentage;
 
     /**
      * @var bool
      */
-    private $appliesToDiscountedProducts;
+    private $excludeDiscountedProducts;
```

Comment:
main idea was to get rid of double negation. I thought having `Exclude discounted products? Yes/No` and avoid double negation was cleaner than having "abit" clearer property name but double negation everwywhere.
So Im not sure which of the cases are you suggesting? because its either double negation as it was, or the change I've done now :smile: (btw in UI it was also called "exclude discounted products")